cpp/src/lance/format/schema.cc

```diff
@@ -499,6 +499,19 @@ ::arrow::Result<std::shared_ptr<Schema>> Schema::Project(
   return projection;
 }
 
+::arrow::Result<std::shared_ptr<Schema>> Schema::Project(
+    const ::arrow::compute::Expression& expr) const {
+  if (!::arrow::compute::ExpressionHasFieldRefs(expr)) {
+    /// All scalar?
+    return nullptr;
+  }
+  std::vector<std::string> columns;
+  for (auto& ref : ::arrow::compute::FieldsInExpression(expr)) {
```

Comment:
This does solve the nested field reference problem right?

Reply:
Oh, no. That's separated.